src/LibFTD2XX.jl

```diff
@@ -1,260 +1,663 @@
-  :FT_CreateDeviceInfoList
-  :FT_GetDeviceInfoList
-  :FT_Open
-  :FT_Close
-  :FT_Read
-  :FT_Write
-  :FT_SetBaudRate
-  :FT_SetDataCharacteristics
-  :FT_GetModemStatus
-  :FT_GetQueueStatus
-  :FT_OpenEx
-  :FT_Purge]
-
-function __init__()
-  lib[] = Libdl.dlopen(libFTD2XX)
-  for n in cfuncn
-    cfunc[n] = Libdl.dlsym(lib[], n)
-  end
-end
 
-# Types
-# 
-struct FT_DEVICE_LIST_INFO_NODE
-  flags::ULONG
-  typ::ULONG
-  id::ULONG
-  locid::DWORD
-  serialnumber::NTuple{16, Cchar}
-  description::NTuple{64, Cchar}
-  fthandle::Ptr{Cvoid}
+"""
+    @enum(
+      FTOpenBy,
+      OPEN_BY_SERIAL_NUMBER = FT_OPEN_BY_SERIAL_NUMBER,
+      OPEN_BY_DESCRIPTION = FT_OPEN_BY_DESCRIPTION)
+
+For use with [`open`](@ref).
+"""
+@enum(
+  FTOpenBy,
+  OPEN_BY_SERIAL_NUMBER = FT_OPEN_BY_SERIAL_NUMBER,
+  OPEN_BY_DESCRIPTION = FT_OPEN_BY_DESCRIPTION)
+
+
+"""
+    @enum(
+      FTWordLength,
+      BITS_8 = FT_BITS_8,
+      BITS_7 = FT_BITS_7)
+
+For use with [`datacharacteristics`](@ref).
+"""
+@enum(
+  FTWordLength,
+  BITS_8 = FT_BITS_8,
+  BITS_7 = FT_BITS_7)
+
+
+"""
+    @enum(
+      FTStopBits,
+      STOP_BITS_1 = FT_STOP_BITS_1,
+      STOP_BITS_2 = FT_STOP_BITS_2)
+
+For use with [`datacharacteristics`](@ref).
+"""
+@enum(
+  FTStopBits,
+  STOP_BITS_1 = FT_STOP_BITS_1,
+  STOP_BITS_2 = FT_STOP_BITS_2)
+
+"""
+    @enum(
+      FTParity,
+      PARITY_NONE = FT_PARITY_NONE,
+      PARITY_ODD  = FT_PARITY_ODD,
+      PARITY_EVEN = FT_PARITY_EVEN,
+      PARITY_MARK = FT_PARITY_MARK,
+      PARITY_SPACE = FT_PARITY_SPACE)
+
+For use with [`datacharacteristics`](@ref).
+"""
+@enum(
+  FTParity,
+  PARITY_NONE = FT_PARITY_NONE,
+  PARITY_ODD  = FT_PARITY_ODD,
+  PARITY_EVEN = FT_PARITY_EVEN,
+  PARITY_MARK = FT_PARITY_MARK,
+  PARITY_SPACE = FT_PARITY_SPACE)
+
+
+"""
+    D2XXException <: Exception
+
+LibFTD2XX High-Level Library Error Type.
+"""
+struct D2XXException <: Exception
+  str::String
 end
 
-mutable struct FT_HANDLE<:IO 
-  p::Ptr{Cvoid} 
+
+"""
+    struct D2XXDevice <: IO
+
+Device identifier for a D2XX device.
+
+See also: [`D2XXDevices`](@ref), [`deviceidx`](@ref), [`deviceflags`](@ref), 
+[`devicetype`](@ref), [`deviceid`](@ref), [`locationid`](@ref), 
+[`serialnumber`](@ref), [`description`](@ref), [`fthandle`](@ref).
+"""
+struct D2XXDevice <: IO
+  idx::Int
+  flags::Int
+  typ::Int
+  id::Int
+  locid::Int
+  serialnumber::String
+  description::String
+  fthandle::Ref{FT_HANDLE}
 end
 
-function FT_HANDLE()
-  handle = FT_HANDLE(C_NULL)
-  @compat finalizer(destroy!, handle)
-  handle
+
+# D2XXDevice Constructors
+#
+"""
+    D2XXDevice(deviceidx::Integer)
+
+Construct a `D2XXDevice` without opening it. D2XX hardware must pre present to 
+work.
+"""
+function D2XXDevice(deviceidx::Integer)
+  0 <= deviceidx || throw(DomainError("0 <= deviceidx"))
+  idx, flags, typ, id, locid, serialnumber, description, fthandle = getdeviceinfodetail(deviceidx)
+  D2XXDevice(idx, flags, typ, id, locid, serialnumber, description, fthandle)
```

Comment:
Just splat it

```
D2XXDevice(getdeviceinfodetail(deviceidx)...)
```

Reply:
resolved in b701a0b9fe82bdab53bf0c02a774d4dc99d3a382